### PR TITLE
Add refactoring commands for clojure

### PR DESCRIPTION
**Pull request by ProjectFrank**
Most of these refactoring commands can be used via `lsp-execute-code-action`, but some of them, such as `move-to-let`, are interactive so `clojure-lsp` does not provide the full set of arguments in the code actions response.

Regardless, this PR contains commands for all of the `clojure-lsp` refactoring operations currently supported.

### CODE REVIEW

File: lsp-clojure.el [hunk 1]
```diff
@@ -37,6 +37,83 @@
...
   :type 'list)
 
+;; Refactorings
+
+(defun lsp-clojure--refactoring-call (refactor-name &rest additional-args)
+  "Send an executeCommand request for REFACTOR-NAME with ADDITIONAL-ARGS if there are more arguments expected after the line and column numbers."
+  (lsp--cur-workspace-check)
+  (lsp--send-execute-command
+   refactor-name
+   (append (list (concat "file://" buffer-file-name)
```
Comment (yyoncho) [inline lsp-clojure.el]: use lsp--path-to-uri

Comment (yyoncho) [reply lsp-clojure.el]: ... and you can use `(list*` instead of `(append (list... )  (list ..))`

Comment (ProjectFrank) [reply lsp-clojure.el]: done!

**Comment by yyoncho** [comment]
Thank you for contributing to `lsp-mode`!

And congrats for your first `lsp-mode` PR! :tada:

**Comment by ProjectFrank** [comment]
thanks!

**Comment by benedekfazekas** [comment]
this looks nice. will deffo give it a spin

**Pull request merged**